src/java.desktop/share/classes/javax/swing/plaf/basic/BasicTextUI.java

```diff
@@ -2610,6 +2610,7 @@ public boolean importData(JComponent comp, Transferable t) {
                     }
 
                     handleReaderImport(r, c, useRead);
+                    r.close();
```

Comment:
Should we use try-with-resources for Reader? This will ensure the stream is closed even if `handleReaderImport` throws an exception.

Reply:
Seems like good idea..Modified...